### ISSUE TITLE
[BUG] Fixed header doesn't match because of other parameters

### ISSUE THREAD

**Issue by ghostdogpr** [softwaremill/tapir#1582]
Tapir version: 0.19.0-M13

Scala version: 2.13

**Describe the bug**

I set a fixed header input `in(header(Header.contentType(MediaType.MultipartFormData)))` to match `multipart/form-data` content type. But I am getting a `UnsupportedMediaType` error because it compares `multipart/form-data` with the actual input that is `multipart/form-data; boundary=EZFJtFg92mYYTWNqBt70vKcuCdW3r563tOsBJrb`.

Is that intended? I know that sometimes `encoding=utf-8` is also appended at the end the content-type, so it feels like using a fixed header will not work reliably. Did I miss something? 😄

**Comment by ghostdogpr** [comment]
Workaround:
```scala
.in(header[String](HeaderNames.ContentType).mapDecode { h =>
  DecodeResult
    .fromEitherString(h, MediaType.parse(h))
    .flatMap(mediaType =>
      if (mediaType.mainType == "multipart" && mediaType.subType == "form-data") DecodeResult.Value(())
      else DecodeResult.Mismatch("multipart/form-data", s"${mediaType.mainType}/${mediaType.subType}")
    )
  }(_ => "multipart/form-data"))
```
But I still wonder if it's indended or if there is a better way.

**Comment by adamw** [comment]
@ghostdogpr yeah, checking `boundary` is a bug, though checking `charset` is correct I think ... though I'll have to verify that.

Another question is - more out of curiosity - why do you need to explicitly set the content-type header. Maybe there's some use-case that we could do better :)

**Comment by ghostdogpr** [comment]
The graphql protocol is a little different when `application/graphql` is sent by the client. We had [this bug](https://github.com/ghostdogpr/caliban/pull/1124) fixed recently (not using Tapir) where we checked the whole string and it didn't match because of the charset. And I noticed the same thing was happening with Tapir if I used a fixed header to catch those calls.

The other case `header(Header.contentType(MediaType.MultipartFormData))` is maybe not necessary, but if you only use `.in(multipartBody[Book])` it doesn't check the header, right? I thought it was better to enforce it.
